DNC af quest

ProfessorCattington opened the pull request:
First pass at the rest of the DNC AF quests. Mostly .lua changes and a small change to the lua_baseentity class.

teschnei commented on scripts/zones/Bastok_Markets/npcs/Matthias.lua:
```diff
@@ -15,17 +15,125 @@ require("scripts/zones/Bastok_Markets/TextIDs");
+            local dancerAFID = 1; -- variable used to convert player's choice into an Item ID. gender is actually important here because it displays the item on screen for you
+            local playerGender = player:getGender();
+			
+            switch (playersAFChoice) : caseof{
```
just use a if/elseif chain, I'm going to deprecate the switch eventually (it doesn't perform better like it does in C/C++)

xdemolish replied:
stick a space between ')' and 'then', and remove any extra ' ' before the closing ')'
also, stick a space between if and parentheses so it looks like this
```
if (player:getFreeSlotsCount() == 0) then
```

oscsoftware commented:
:+1:

teschnei commented on src/map/lua/lua_baseentity.cpp:
```diff
@@ -8734,6 +8734,17 @@ inline int32 CLuaBaseEntity::isSpellAoE(lua_State* L)
+inline int32 CLuaBaseEntity::getGender(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_NPC);
```
since you're casting it to a CharEntity, you should be checking if it's not a TYPE_PC

teschnei commented on scripts/zones/Bastok_Markets/npcs/Matthias.lua:
```diff
@@ -15,17 +15,125 @@ require("scripts/zones/Bastok_Markets/TextIDs");
+		
+        if(player:getFreeSlotsCount() == 0) then
+		
+            player:PrintToPlayer( "Not enough free inventory space." );
```
Quests should never be using PrintToPlayer, since retail never ever does this (it's always message packets)

teschnei commented:
nah, you just push to this branch again and then post here that you did (since there's no notification for a new commit, just for comments)

ProfessorCattington commented:
Ah OK. I thought I already pushed.

teschnei commented:
you did :)

ProfessorCattington commented:
I did it! :3

teschnei commented:
your editor replaced a bunch of trailing whitespace with tabs for some reason (check harmodios's diff to see what I mean), and there are a few tabs hanging around instead of spaces (matthias, for example).  if you could a) remove trailing whitespace and b) convert tabs to 4 spaces, that'd be good

teschnei commented on scripts/zones/Bastok_Markets/npcs/Matthias.lua:
```diff
@@ -46,7 +161,64 @@ end;
+        end;
+		
+        if(player:getFreeSlotsCount() == 0) then --check to see if the player has enough inventory space before rewarding them.
+            player:messageSpecial(ITEM_CANNOT_BE_OBTAINED, dancerAFID);
```
if the inventory is full, the function should immediately return - or else you set the var for having completed the quest, but they won't obtain the item

ProfessorCattington replied:
Did you mean just add a return statement after the messageSpecial call or add the return statement and move the block up in the function/event code?

teschnei replied:
just do whatever to make sure that you can't complete the quest without getting the item

ProfessorCattington commented:
I went through the dancer AF scripts and removed a bunch of white space and tabs. Also changed Matthias's script per request.

What editor would you recommend for lua? I've been playing around with some I found online but I don't really like any of them and notepad++ is a bit limited.

helixhamin commented:
Sadly, I would still use Notepad++, I cannot think of much else that I would recommend.
What limitations are you seeing?
btw, to auto remove trailing whitespace, save with ctrl+s (only if you are using notepad++)
It shows under the "Macro" menu at the top.

teschnei commented:
I use vim, but notepad++ and sublimetext for example would still be fine.
Just change the settings to insert spaces instead of tabs and it'll be fine

On Tue, Feb 2, 2016 at 10:02 PM, helixhamin <notifications@github.com>
wrote:

> Sadly, I would still use Notepad++, I cannot think of much else that I
> would recommend.
> What limitations are you seeing?
> btw, to auto remove trailing whitespace, save with ctrl+s (only if you are
> using notepad++)
> It shows under the "Macro" menu at the top.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DarkstarProject/darkstar/pull/2659#issuecomment-179010654>
> .
>

helixhamin commented:
For the settings, to replace tab, select Settings -> Preferences -> Tab Settings -> make sure box showing "Replace by space" is checked.

xdemolish commented:
space between if/elseif and parentheses :D

aizenmyou commented:
Eclipse has a Lua plugin with autocompletion (can be a mixed blessing), though I have not tried it as I'm in the Vim crowd for things like scripts.

ProfessorCattington commented:
It would be nice to see and jump to references. I'm going to take a look at vim and the eclipse plugin. Thanks everyone.

ProfessorCattington commented:
placed spaces between if/elseif and parenthesis.

xdemolish commented:
thanks!
